comms/discovery/pubkeystore/backfill.go

```diff
@@ -49,3 +60,44 @@ func setPubkey(userId int, pubkeyBase64 string) error {
 	_, err := db.Conn.Exec(`insert into user_pubkeys values ($1, $2) on conflict do nothing`, userId, pubkeyBase64)
 	return err
 }
+
+func recoverFromPeers(discoveryConfig *config.DiscoveryConfig, id int) error {
+	idEncoded, err := misc.EncodeHashId(id)
+	if err != nil {
+		return err
+	}
+	for _, peer := range discoveryConfig.Peers() {
+		if strings.EqualFold(peer.Wallet, discoveryConfig.MyWallet) {
+			continue
+		}
+
+		host := strings.TrimRight(peer.Host, "/")
+		if host == "" {
+			continue
+		}
+
+		resp, err := http.Get(host + "/comms/pubkey/" + idEncoded + "/cached")
```

Comment:
For my understanding, this means it only checks the pubkeystore cache on the peer, and doesn't do the lookup attempt against chain?

Reply:
Yeah with the `/cached` route suffix... the peer will only look in db... so it won't cause unwanted cascading request amplification stuff.

Reply:
The current host will first try to recover from chain and only fallback to asking a peer for a cached version if that fails.